src/main/java/org/folio/rest/core/RestClient.java

```diff
@@ -1,191 +1,293 @@
-      client.closeClient();
-      future.completeExceptionally(e);
+    Promise<T> promise = Promise.promise();
+    getVertxWebClient(requestContext.getContext())
+      .getAbs(buildAbsEndpoint(caseInsensitiveHeader, endpoint))
+      .putHeaders(caseInsensitiveHeader)
+      .expect(SUCCESS_RESPONSE_PREDICATE)
+      .send()
+      .map(HttpResponse::bodyAsJsonObject)
+      .map(jsonObject -> {
+        if (log.isDebugEnabled()) {
+          log.debug("Successfully retrieved: {}", jsonObject.encodePrettily());
+        }
+        return jsonObject.mapTo(responseType);
+      })
+      .onSuccess(promise::complete)
+      .onFailure(t -> handleGetMethodErrorResponse(promise, t, skipError404));
+
+    return promise.future();
+  }
+
+
+  public Future<JsonObject> getAsJsonObject(String endpoint, boolean skipError404, RequestContext requestContext) {
+    log.debug("Calling GET {}", endpoint);
+    Promise<JsonObject> promise = Promise.promise();
+    var caseInsensitiveHeader = convertToCaseInsensitiveMap(requestContext.getHeaders());
+    var webClient = getVertxWebClient(requestContext.getContext());
+
+    webClient.getAbs(buildAbsEndpoint(caseInsensitiveHeader, endpoint))
+      .putHeaders(caseInsensitiveHeader)
+      .expect(SUCCESS_RESPONSE_PREDICATE)
+      .send()
+      .map(HttpResponse::bodyAsJsonObject)
+      .onSuccess(jsonObject -> {
+        if (log.isDebugEnabled()) {
+          log.debug("Successfully retrieved: {}", jsonObject.encodePrettily());
+        }
+        promise.complete(jsonObject);
+      })
+      .onFailure(t -> handleGetMethodErrorResponse(promise, t, skipError404));
+    return promise.future();
+  }
+
+  public Future<JsonObject> getAsJsonObject(RequestEntry requestEntry, boolean skipError404, RequestContext requestContext) {
+    return getAsJsonObject(requestEntry.buildEndpoint(), skipError404, requestContext);
+  }
+
+  public Future<JsonObject> getAsJsonObject(RequestEntry requestEntry, RequestContext requestContext) {
+    return getAsJsonObject(requestEntry.buildEndpoint(), false, requestContext);
+  }
+
+  public String extractRecordId(HttpResponse<Buffer> response) {
+    JsonObject body = response.bodyAsJsonObject();
+    String id;
+    if (body != null && !body.isEmpty() && body.containsKey(ID)) {
+      id = body.getString(ID);
+    } else {
+      String location = response.getHeader(LOCATION);
+      id = location.substring(location.lastIndexOf('/') + 1);
     }
-    return future;
+    return id;
   }
 
-  protected HttpClientInterface getHttpClient(Map<String, String> okapiHeaders) {
-    final String okapiURL = okapiHeaders.getOrDefault(RestConstants.OKAPI_URL, "");
-    final String tenantId = TenantTool.calculateTenantId(okapiHeaders.get(OKAPI_HEADER_TENANT));
+  protected WebClient getVertxWebClient(Context context) {
+    WebClientOptions options = new WebClientOptions();
+    options.setLogActivity(true);
+    options.setKeepAlive(true);
+    options.setConnectTimeout(2000);
```

Comment:
Default connect timeoup is 60 sec, maybe lets back to original value?
and default idle timeout is 0, you made it to not keep connection active all time and recreated each 5 sec if needed?
and keepAlive set to true to reuse active connection by other threads?
This configuration applied also in other modules?

Reply:
This is the values from default configuration for creating module clients by rmb 
https://github.com/folio-org/raml-module-builder/blob/master/domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java#L192
I decided to keep them the same